actors/src/main/kotlin/ktx/actors/actions.kt

```diff
@@ -34,24 +34,25 @@ operator fun Actor.minusAssign(action: Action) = removeAction(action)
 
 /**
  * Wraps this action and the passed action with a [SequenceAction].
- * @param action will be executed after this action.
- * @return [SequenceAction] storing both actions.
- */
-infix fun Action.then(action: Action): SequenceAction = Actions.sequence(this, action)
-
-/**
- * Wraps the actions in this [SequenceAction] with the passed action in a new [SequenceAction]
  *
- * The underling actions present in this [SequenceAction] will be unwrapped.
+ * Eventual underling actions will be unwrapped.
  *
- * @param action will be executed after this sequence of action.
+ * @param action will be executed after this action.
  * @return [SequenceAction] storing both actions.
  */
-infix fun SequenceAction.then(action: Action): SequenceAction {
-  val sequence = SequenceAction()
-  actions.forEach { sequence += it }
-  sequence += action
-  return sequence
+infix fun Action.then(action: Action): SequenceAction {
+  val result = SequenceAction()
+  result.addUnwrapped(this)
+  result.addUnwrapped(action)
+  return  result
+}
+
+private fun SequenceAction.addUnwrapped(action: Action) {
+  if (action is SequenceAction) {
+    action.actions.forEach(this::addAction)
```

Comment:
Isn't `this` optional in this case?

To you think we should unwrap the actions recursively (`forEach(::addUnwrapped)`)?

Reply:
Ah yes

Reply:
For the record - extracting actions recursively might be excessive and rare in practice.

Reply:
> extracting actions recursively might be excessive and rare in practice.


Yes, which is why I didn't do it in the first place. 

Actually I said "Ah yes" about the "this" being optional.

I'm not a big fan of doing `addUnwrapped` recursively, for the reason you mentionned. Please confirm if you want me to make it recursive.